src/plots/mapbox/layers.js

```diff
@@ -70,6 +84,13 @@ proto.needsNewLayer = function(opts) {
     );
 };
 
+proto.updateImage = function(opts) {
+    var map = this.subplot.map;
+    map.getSource(this.idSource).updateImage({
```

Comment:
This is absolutely brilliant :sparkles: 

Thanks very much! Let's get a test case in to lock down this beautiful optimization.

Reply:
Test added in https://github.com/plotly/plotly.js/pull/4418/commits/4158916421deb440690dd3379a2783419a95112c